lib/utils/request.js

```diff
@@ -0,0 +1,19 @@
+module.exports = request
+
+var nets = require('nets')
+
+var Promise = require('./promise')
+
+function request (options) {
+  options.encoding = undefined
+
+  return new Promise(function (resolve, reject) {
+    nets(options, function (error, response) {
+      if (error) {
+        return reject(error)
+      }
```

Comment:
I'm assuming we just want to pass along the whole response body?

`
      if (response.statusCode >= 400) {
        return reject(new Error(response.body))
      }
`